testStrategies/commonToAllTestStrategies.go

```diff
@@ -63,57 +62,37 @@ type XMLTestDefinition struct {
 }
 
-// TestDefinition encapsulates the test config file.
-type TestDefinition struct {
-	TestName            string               `toml:"testName"`
-	OverrideHost        string               `toml:"overrideHost"`
-	OverridePort        string               `toml:"overridePort"`
-	HTTPMethod          string               `toml:"httpMethod"`
-	BaseURI             string               `toml:"baseUri"`
-	Multipart           bool                 `toml:"multipart"`
-	Payload             string               `toml:"payload"`
-	MultipartPayload    []multipartFormField `toml:"multipartFormField"`
-	ResponseStatusCode  int                  `toml:"responseStatusCode"`
-	ResponseContentType string               `toml:"responseContentType"`
-	Headers             http.Header          `toml:"headers"`
-	ResponseValues      []ResponseValue      `toml:"responseProperties"`
-	// Attributes defined in the testSuite:
-	PreThinkTime  int64
-	PostThinkTime int64
-	ExecWeight    string
+// TestSuite fields get populated from the TestSuiteDefinition after the XML
+// unmarshal is complete. (See TestDefinition above).
+type TestSuite struct {
+	XMLName         xml.Name   `xml:"testSuite"`
+	Name            string     `xml:"name"`
+	TestStrategy    string     `xml:"testStrategy"`
+	TestCases       []TestCase `xml:"testCases>testCase"`
+	TestDefinitions []*TestDefinition
 }
 
-// TestCase defines a load test scenario.
+// TestCase is used to encapsulate and marshal a <testCase> tag from the
+// <testSuite> XML file. This data will then be consolidated into
+// the TestSuite/TestDefinition data structure for usage.
 type TestCase struct {
-	Name          string `xml:",chardata"`
-	PreThinkTime  int64  `xml:"preThinkTime,attr"`
-	PostThinkTime int64  `xml:"postThinkTime,attr"`
-	ExecWeight    string `xml:"execWeight,attr"`
-}
-
-// TestSuiteDefinition encapsulates a load test scenario.
-type TestSuiteDefinition struct {
-	XMLName      xml.Name   `xml:"testSuite"`
-	Name         string     `xml:"name" toml:"name"`
-	TestStrategy string     `xml:"testStrategy" toml:"testStrategy"`
-	TestCases    []TestCase `xml:"testCases>testCase" toml:"testCases"`
-}
-
-// TestSuite defines a load test scenario.
-type TestSuite struct {
-	XMLName      xml.Name
-	Name         string
-	TestStrategy string
-	TestCases    []*TestDefinition
+	XMLName       xml.Name `xml:"testCase"`
+	Name          string   `xml:",chardata"`
+	PreThinkTime  int64    `xml:"preThinkTime,attr"`
+	PostThinkTime int64    `xml:"postThinkTime,attr"`
+	ExecWeight    string   `xml:"execWeight,attr"`
 }
 
 type multipartFormField struct {
-	FieldName   string `xml:"fieldName" toml:"fieldName"`
-	FieldValue  string `xml:"fieldValue" toml:"fieldValue"`
-	FileName    string `xml:"fileName" toml:"fileName"`
-	FileContent []byte `xml:"fileContent" toml:"fileContent"`
+	FieldName   string `xml:"fieldName"`
+	FieldValue  string `xml:"fieldValue"`
+	FileName    string `xml:"fileName"`
+	FileContent []byte `xml:"fileContent"`
 }
```

Comment:
Changes to simplify the data structures used to unmarshal the XML , since we no longer have to consider TOML. It's hard to tell what's going on from the diff. It might be more clear to look at the final state of the source code.